gym_xiangqi/envs/xiangqi_env.py

```diff
@@ -318,7 +318,7 @@ def step_user(self):
 
         # Process the piece movement in env
         player_action = move_to_action_space(piece_id, start, end)
-        return self.step(player_action)
+        return self.step(int(player_action))
```

Comment:
Oh wouldn't `player_action` be always int anyway? Hmm

Reply:
Yeah I did not get a chance to take a deeper look but for some reason it was `float` and was causing an assert error. I will take a closer look again with Danny.

Reply:
Yea, it worked just fine before I messed with the coordinate conversion methods in pygame, which I don't think is much related to this. Let's look more together soon.

Reply:
I found that the coordinates are unintentionally returned as `float`. We can fix it in the `to_real_coor()` method.